src/devices/Button/GpioButton.cs

```diff
@@ -75,7 +101,7 @@ private void PinStateChanged(object sender, PinValueChangedEventArgs pinValueCha
             switch (pinValueChangedEventArgs.ChangeType)
             {
                 case PinEventTypes.Falling:
-                    if (_pinMode == PinMode.InputPullUp)
+                    if (_eventPinMode == PinMode.InputPullUp)
```

Comment:
same here, you can't really do it like this is pull up is not supported

Reply:
The `_eventPinMode` is always good because it is used to keep track if it is either pulled-up internally or externally.
The validation only goes when I open the pin. But if the resistor is external, the raising/falling logic is basically the same.

Reply:
To make this more clear: `_eventPinMode` is **not** used in `OpenPin`. 
I just 'reused" the enumeration to keep track whether the button transition is high to low or vice-versa.